docs/DevGuide.rst

```diff
@@ -147,20 +147,50 @@ We want to ensure that code across NetworKit is easy to understand for existing
 -  Document classes, methods and attributes in Doxygen style.
 -  Use the ``count`` and ``index`` integer types for non-negative
    integer quantities and indices.
+-  All member variables should be pointers and not references.
 -  In most cases, objects are passed by reference. New objects are
-   stack-allocated and returned by value. Avoid pointers and ``new``
+   stack-allocated and returned by value. Avoid ``new``
    where possible.
 -  Use the ``override`` keyword to indicate that a method overrides a
    virtual method in the superclass.
+-  A class should be declared ``final`` unless it is a superclass.
+-  ``virtual`` methods should only be declared in superclasses.
 -  In Python, indent using tabs, not spaces.
 
+In order to maintain the same standard of code across the entire NetworKit code base, some coding standards are enforced. However, there is some automation to help developers with this. Below is a list of these standards and instructions on how to use the available automation tools that ensure your code adheres to them.
+
+-  ``CppClangFormat`` applies clang-format to all files that contain the
+   string ``networkit-format``.
+    After adding a new file to the code base, subscribe the file to ``networkit-format`` by adding the string "networkit-format" to the file after the include guards as is shown below.
+    ::
+
+        /*
+        * GlobalCurveballImpl.hpp
+        ...
+        */
+        // networkit-format
+
+-  ``CppIndentation`` checks that all C++ code is indented with spaces
+   and not tabs.
+-  ``CppIncludeGuards`` ensures that the header files contain an include guard and
+   that it follows the following naming convention: ``NETWORKIT_MODULENAME_HEADERFILE_HPP_``.
+
+The executable file ``check_code.sh`` carries out all checks in read-only mode and reports if errors are found. Running ``check_code.sh -w`` will fix these errors. Run this script before commiting your files to make sure your changes are in complaince with the guidelines. The script is executed during CI and will cause your pull request to fail if your code does not conform to the style guide.
```

Comment:
Please include a reference to the folder of check_code.sh.

Reply:
Done.